pkg/ios/dsym-utils.go

```diff
@@ -0,0 +1,138 @@
+package ios
+
+import (
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/pkg/errors"
+
+	"github.com/bugsnag/bugsnag-cli/pkg/log"
+	"github.com/bugsnag/bugsnag-cli/pkg/utils"
+)
+
+// DwarfInfo stores the UUID, architecture and name of a dwarf file
+type DwarfInfo struct {
+	UUID     string
+	Arch     string
+	Name     string
+	Location string
+}
+
+func FindDsymsInPath(path string) ([]*DwarfInfo, string, error) {
+	var tempDir string
+	var dsymLocations []string
+	var dwarfInfo []*DwarfInfo
+
+	// If path is set and is a directory
+	if utils.IsDir(path) {
+		// Check for dSYMs within it
+		dsymLocations = findDsyms(path)
+
+	} else {
+
+		// If path is pointing to a .zip file, we will extract it and look for dSYMS within it to get dsymLocations
+		if strings.HasSuffix(path, ".zip") {
+
+			fileName := filepath.Base(path)
+			log.Info("Attempting to unzip " + fileName + " before proceeding to upload")
+
+			var err error
+			tempDir, err = utils.ExtractFile(path, "dsym")
+
+			if err != nil {
+				// TODO: This will be downgraded to a warning with --fail-on-upload in near future
+				log.Error("Could not unzip "+fileName+" to a temporary directory, skipping", 1)
+			} else {
+				log.Info("Unzipped " + fileName + " to " + tempDir + " for uploading")
+				dsymLocations = findDsyms(tempDir)
+
+			}
+
+		} else if strings.HasSuffix(path, ".dSYM") {
```

Comment:
General thought: is it worth considering case-insensitive checks when doing these? Especially given how inconsistent things get between platforms?

Reply:
.zip more likely than .dSYM for sure, but calling toLower for all these cases seems like a good shout.